langchain4j-elasticsearch/src/main/java/dev/langchain4j/store/embedding/elasticsearch/ElasticsearchEmbeddingStore.java

```diff
@@ -367,6 +399,50 @@ private void bulk(List<String> ids, List<Embedding> embeddings, List<TextSegment
         }
     }
 
+    private void removeByQuery(Query query) {
+        try {
+            DeleteByQueryResponse response = client.deleteByQuery(delete -> delete
+                    .index(indexName)
+                    .query(query));
+            if (!response.failures().isEmpty()) {
+                for (BulkIndexByScrollFailure item : response.failures()) {
+                    if (item.cause() != null) {
+                        throw new ElasticsearchRequestFailedException("type: " + item.cause().type() + ", reason: " + item.cause().reason());
+                    }
+                }
+            }
+        } catch (IOException e) {
+            log.error("[ElasticSearch encounter I/O Exception]", e);
```

Comment:
no need to log if exception is thrown further

Reply:
also removed the log from existing methods in this class